builtin/ui/window.lua

```diff
@@ -0,0 +1,308 @@
+--[[
+Minetest
+Copyright (C) 2023 v-rob, Vincent Robinson <robinsonvincent89@gmail.com>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+--]]
+
+local open_windows = {}
```

Comment:
What will happen when users exit or disconnect from game with some UI opened, e.g. close window after opening some UI? Will the UI be reopened on the next time user re-join game?

Reply:
Strange; I was certain that I had a `core.register_on_leaveplayer` to remove their windows from the table, but it must have been removed at some point. However, the expected behavior is that the window is closed when the player leaves and is not reopened if they come back.